src/synapse/azext_synapse/custom.py

```diff
@@ -121,59 +123,126 @@ def create_workspace(cmd, client, resource_group_name, workspace_name, account_u
 
 
 def update_workspace(cmd, client, resource_group_name, workspace_name, sql_admin_login_password=None,
-                     identity_type="SystemAssigned", principal_id=None, tags=None, no_wait=False):
-    identity = ManagedIdentity(type=identity_type)
-    workspace_patch_info = WorkspacePatchInfo(
-        tags=tags,
-        identity=identity,
-        sql_administrator_login_password=sql_admin_login_password
-    )
+                     tags=None, no_wait=False):
+    workspace_patch_info = WorkspacePatchInfo(tags=tags, sql_admin_login_password=sql_admin_login_password)
     return sdk_no_wait(no_wait, client.update, resource_group_name, workspace_name, workspace_patch_info)
 
 
+def custom_check_name_availability(cmd, client, name):
+    return client.check_name_availability(name, "Microsoft.Synapse/workspaces")
+
+
 def get_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name):
     return client.get(resource_group_name, workspace_name, spark_pool_name)
 
 
-def create_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name, location,
-                      spark_version, node_size=NodeSize.medium.value, node_count=3,
-                      node_size_family=NodeSizeFamily.memory_optimized.value, auto_scale_enabled=True,
-                      min_node_count=3,
-                      max_node_count=40, auto_pause_enabled=True, delay_in_minutes=15, spark_events_folder="/events",
-                      library_requirements_filename=None, library_requirements_content=None,
+def create_spark_pool(cmd, client, resource_group_name, workspace_name, spark_pool_name,
+                      spark_version, node_size=NodeSize.medium.value,
+                      node_size_family=NodeSizeFamily.memory_optimized.value, enable_auto_scale=None,
+                      node_count=None, min_node_count=None, max_node_count=None,
+                      enable_auto_pause=None, delay_in_minutes=None, spark_events_folder="/events",
+                      library_requirements_file=None,
                       default_spark_log_folder="/logs", force=False, tags=None, no_wait=False):

```

Comment:
>force=False [](start = 56, length = 11)

why do we need a `force` parameter here?

Reply:
removed

---
In reply to: [417135964](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r417135964) [](ancestors = 417135964)